torch/_refs/__init__.py

```diff
@@ -4673,10 +4673,50 @@ def full_like(
     return fill(e, fill_value)
 
 
-zeros_like = partial(full_like, fill_value=False)
+@register_decomposition(aten.zeros_like)
+def zeros_like(
+    a: TensorLikeType,
+    *,
+    dtype: Optional[torch.dtype] = None,
+    layout: Optional[torch.layout] = None,
+    device: Optional[torch.device] = None,
+    pin_memory: bool = False,
+    requires_grad: bool = False,
+    memory_format: torch.memory_format = torch.preserve_format,
+) -> TensorLikeType:
+    return torch.full_like(
+        a,
+        False if (dtype or a.dtype) == torch.bool else 0,
```

Comment:
Why doesn't `False` work here like it did with `_refs.full_like`?

Reply:
It does, but the `False` did show in the triton code and it was a bit odd. This way it should be a bit more type-safe.

Reply:
That sound like an inductor bug then, I would expect it to convert the literal to the tensor dtype.

Reply:
Fix for this in #92093

Reply:
fantastic, thank you!